File: scripts/generate_key_hash.py
#!/usr/bin/env python3
"""
scripts/generate_key_hash.py
─────────────────────────────────────────────────────────────────────────────
Utility for rotating the Developer Preview access key.

Usage:
    python scripts/generate_key_hash.py <new-access-key>

Output:
    The SHA-256 hex digest of the key, ready to paste into:
        backend/core/access.py → EXPECTED_KEY_HASH

This script has no dependencies beyond Python stdlib.
Requires Python 3.6+.
"""

import hashlib
import sys


def main() -> None:
    if len(sys.argv) < 2 or not sys.argv[1].strip():
        print("Usage: python scripts/generate_key_hash.py <access-key>", file=sys.stderr)
        sys.exit(1)

    key = sys.argv[1].strip()
    digest = hashlib.sha256(key.encode("utf-8")).hexdigest()

    border = "─" * 60
    print(f"\n{border}")
    print(f"  Key     : {key}")
    print(f"  SHA-256 : {digest}")
    print(f"{border}")
    print()
    print("  Paste into backend/core/access.py → EXPECTED_KEY_HASH:\n")
    print(f'  EXPECTED_KEY_HASH: str = "{digest}"')
    print()
    print("  Then update BAKUP_ACCESS_KEY in your .env file.\n")


if __name__ == "__main__":
    main()

File: scripts/generate-key-hash.js
/**
 * Usage: node scripts/generate-key-hash.js <access-key>
 *
 * Outputs the SHA-256 hex hash of the given key.
 * Copy the result into VALID_KEY_HASH in app.js.
 *
 * This script has no dependencies. Node.js 15+ required (built-in crypto).
 */
const { createHash } = require('crypto');

const key = process.argv[2];

if (!key) {
  console.error('Usage: node scripts/generate-key-hash.js <access-key>');
  process.exit(1);
}

const hash = createHash('sha256').update(key.trim(), 'utf8').digest('hex');

console.log('\nKey     :', key);
console.log('SHA-256 :', hash);
console.log('\nPaste into app.js → VALID_KEY_HASH:\n');
console.log(`const VALID_KEY_HASH = '${hash}';`);
